src/Riak/Command/Indexes/Query.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+/*
+Copyright 2014 Basho Technologies, Inc.
+
+Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+namespace Basho\Riak\Command\Indexes;
+
+use Basho\Riak\Command;
+use Basho\Riak\CommandInterface;
+
+/**
+ * Class Query
+ *
+ * Riak 2i query information.
+ *
+ * @author Alex Moore <amoore at basho d0t com>
+ */
+class Query extends Command\Object implements CommandInterface
+{
+    /**
+     * @var string
+     */
+    protected $indexName = NULL;
+
+    protected $match = NULL;
+    protected $lowerBound = NULL;
+    protected $upperBound = NULL;
+
+    protected $continuation = NULL; //Binary
+    protected $maxResults; // Int
+    protected $returnTerms; // Bool
+    protected $paginationSort; // Bool
+    protected $termFilter; // string
+    protected $timeout; // timeout
+
+    public function __construct(Command\Builder\QueryIndex $builder)
+    {
+        parent::__construct($builder);
+
+        $this->bucket = $builder->getBucket();
+        $this->indexName = $builder->getIndexName();
+
+        if($builder->isRangeQuery()) {
+            $this->lowerBound = $builder->getLowerBound();
+            $this->upperBound = $builder->getUpperBound();
+        }
+        else {
+            $this->match = $builder->getMatchValue();
+        }
+
+        $this->continuation = $builder->getContinuation();
+        $this->maxResults = $builder->getMaxResults();
+        $this->returnTerms = $builder->getReturnTerms();
+        $this->paginationSort = $builder->getPaginationSort();
+        $this->termFilter = $builder->getTermFilter();
+        $this->timeout = $builder->getTimeout();
+    }
+
+    public function getIndexName() {
+        return $this->indexName;
+    }
+
+    public function getMatchValue() {
+        return $this->match;
+    }
+
+    public function getLowerBound() {
+        return $this->lowerBound;
+    }
+
+    public function getUpperBound() {
+        return $this->upperBound;
+    }
+
+    public function isMatchQuery()
+    {
+        return isset($this->match);
+    }
+
+    public function isRangeQuery()
+    {
+        return isset($this->lowerBound) && isset($this->upperBound);
+    }
```

Comment:
For isMatchQuery & isRangeQuery, you already evaluate these expressions within the builder, I would just transfer the result of the expression to properties of the object and set them within the constructor. This reduces the number of PHP expressions that need to be evaluated during runtime.